domains/auth/Auth.API/Unit.Tests/Controllers/OidcControllerTests.cs

```diff
@@ -58,698 +64,563 @@ public OidcControllerTests()
+        new List<object[]>
+        {
+            new object[] {DiscoveryDocument.Load(new List<KeyValuePair<string, string>>() { new("error", "it went all wrong") })},
+            new object[] {null}
+        };
 
     [Theory]
-    [InlineData(null)]
-    [InlineData("/testpath")]
-    public async Task CallbackAsync_ShouldReturnRedirectToOverridenUri_WhenConfigured(string? redirectionPath)
+    [MemberData(nameof(WrongDiscoveryDocumentResponse))]
+    public void DiscoveryDocumentErrorChecks_ShouldThrowAndReturnRedirectUrlWithErrorAndLogWarning_WhenGivenErrorConditions(DiscoveryDocumentResponse? document)
     {
-        var testOptions = TestOptions.Oidc(oidcOptions, allowRedirection: true);
-        var tokenEndpoint = new Uri($"http://{oidcOptions.AuthorityUri.Host}/connect/token");
+        var result = Assert.Throws<OidcException>(() => oidcHelper.DiscoveryDocumentErrorChecks(document, logger, "https://test.dk")).Url;
 
-        var document = DiscoveryDocument.Load(
-            new List<KeyValuePair<string, string>>() {
-                new("issuer", $"https://{testOptions.AuthorityUri.Host}/op"),
-                new("token_endpoint", tokenEndpoint.AbsoluteUri),
-                new("end_session_endpoint", $"http://{testOptions.AuthorityUri.Host}/connect/endsession")
-            },
-            KeySetUsing(tokenOptions.PublicKeyPem)
-        );
-
-        cache.GetAsync().Returns(document);
+        Assert.NotNull(result);
+        Assert.Contains($"{ErrorCode.QueryString}={ErrorCode.AuthenticationUpstream.DiscoveryUnavailable}", result);
 
-        var providerId = Guid.NewGuid().ToString();
-        var name = Guid.NewGuid().ToString();
-        var identityToken = TokenUsing(tokenOptions, document.Issuer, testOptions.ClientId);
-        var accessToken = TokenUsing(tokenOptions, document.Issuer, testOptions.ClientId, claims: new() {
-            { "scope", "something" },
-        });
-        var userToken = TokenUsing(tokenOptions, document.Issuer, testOptions.ClientId, claims: new() {
-            { "mitid.uuid", providerId },
-            { "mitid.identity_name", name },
-            { "idp", ProviderName.MitId  },
-            { "identity_type", ProviderGroup.Private}
-        });
+        logger.Received(1).Log(
+            Arg.Is<LogLevel>(logLevel => logLevel == LogLevel.Error),
+            Arg.Any<EventId>(),
+            Arg.Any<object>(),
+            Arg.Any<Exception>(),
+            Arg.Any<Func<object, Exception?, string>>());
+    }
 
-        userProviderService.GetNonMatchingUserProviders(Arg.Any<List<UserProvider>>(), Arg.Any<List<UserProvider>>()).Returns(new List<UserProvider>());
+    [Fact]
+    public void DiscoveryDocumentErrorChecks_ShouldNotThrowWhen_WhenGivenCorrectConditions() => Assert.Null(Record.Exception(() => oidcHelper.DiscoveryDocumentErrorChecks(DiscoveryDocument.Load(new List<KeyValuePair<string, string>>() { new("token_endpoint", "https://test.dk") }), logger, "https://test.dk")));
```

Comment:
Consider `DoesNotThrow` extension.

Reply:
seems like its common practice to use Record.Exception